signalfx-java/src/main/java/com/signalfx/signalflow/StreamMessage.java

```diff
@@ -0,0 +1,129 @@
+package com.signalfx.signalflow;
+
+/**
+ * Base class for stream messages received from a SignalFlow computation
+ */
+public class StreamMessage {
+
+    /**
+     * Enumeration of kinds of stream messages 
+     */
+    public static enum Kind {
+
+        CONTROL("control-message"),
+        INFORMATION("message"),
+        METADATA("metadata"),
+        DATA("data"),
+        EVENT("event"),
+        ERROR("error");
+
+        private final String specName;
+
+        Kind(String specName) {
+            this.specName = specName;
+        }
+
+        public String toString() {
+            return this.specName;
+        }
+    };
+
+    private String event;
+    private String id;
+    private String data;
+    private Kind kind;
+
+    public StreamMessage() {
+        this.event = "message";
+        kind = Kind.INFORMATION;
+    }
+
+    public StreamMessage(String event, String id, String data) {
+        this.event = event;
+        this.id = id;
+        this.data = data;
+
+        if (isKind(Kind.CONTROL)) {
```

Comment:
Then in your `Kind` enum, statically build a `Map<String, Kind>` and add a static method `findByEventName(String event)` that looks up that map.